src/OrchardCore.Modules/OrchardCore.ContentLocalization/Controllers/AdminController.cs

```diff
@@ -82,6 +82,12 @@ public async Task<IActionResult> Localize(string contentItemId, string targetCul
             {
                 var newContent = await _contentLocalizationManager.LocalizeAsync(contentItem, targetCulture);
                 await _notifier.InformationAsync(H["Localized version of the content created successfully."]);
+
+                if(!string.IsNullOrEmpty(returnUrl))
+                {
+                    TempData["returnUrl"] = returnUrl;
```

Comment:
Shouldn't it be part of the redirect url instead? added as an argument in the route.

Reply:
I I simplified returnUrl implementation
commit: 771d500f16db2ab3fad239ef79a311f1125416d9